### PR TITLE
Add audio duration to L2A button

### PR DESCRIPTION
**Pull request by vlbee**
## What does this change?

This adds the audio duration to the Listent to Article button if it exists and is greater than 0 seconds.
The duration will not show if it's longer than 1hr. 

This PR depends on Bridget v8.6.0 which adds the .getAudioDurationSeconds method 

Figma: https://www.figma.com/design/bIOvOoy1lz8b6X3wbDI1Z2/In-article-changes?node-id=5879-2223&t=w8DzDVesw4pI6gUA-0

## Why?

Part of work to move the L2A button into the body of the article from the header to test that it will increase click rates. 

## Screenshots

image
image

No duration or duration data with errors: 

image
image

[before]: https://example.com/before.png
[after]: https://example.com/after.png

### CODE REVIEW

File: dotcom-rendering/src/components/ListenToArticle.test.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { formatAudioDuration } from './ListenToArticle.importable';
+
+describe('format Audio Duration correctly', () => {
+	const testCases = [
+		{ input: -60, expected: '00:00' },
```
Comment (JamieB-gu) [inline dotcom-rendering/src/components/ListenToArticle.test.tsx]: What's the reason for a negative duration to be `00:00`? Should it instead be empty, as for the large duration, since negative durations presumably aren't valid here?

File: dotcom-rendering/src/components/ListenToArticle.importable.tsx [hunk 2]
```diff
@@ -7,26 +7,55 @@ import { ListenToArticleButton } from './ListenToArticleButton';
...
+
+	if (hours > 3) {
+		return '';
+	}
+
+	const formattedDuration = `${
+		hours > 0 ? `${hours.toString().padStart(2, '0')}:` : ''
+	}${minutes > 0 ? minutes.toString().padStart(2, '0') : '00'}:${
+		seconds > 0 ? seconds.toString().padStart(2, '0') : '00'
+	}`;
```
Comment (JamieB-gu) [inline dotcom-rendering/src/components/ListenToArticle.importable.tsx]: Could this be:

```suggestion
	}${minutes.toString().padStart(2, '0')}:${
		seconds.toString().padStart(2, '0')
	}`;
```

given that the padding code will take care of `'00'` when the values are `0`?

That's assuming negative values aren't valid, and should therefore perhaps result in `''`/`undefined`, as with large durations?

File: dotcom-rendering/src/components/ListenToArticle.importable.tsx [hunk 3]
```diff
@@ -7,26 +7,55 @@ import { ListenToArticleButton } from './ListenToArticleButton';
...
-export const ListenToArticle = ({ articleId }: Props) => {
-	const [showButton, setShowButton] = useState<boolean>(false);
 
-	const isBridgetCompatible = useIsBridgetCompatible('8.5.1');
+export const formatAudioDuration = (durationInSeconds: number): string => {
+	const hours = Math.floor(durationInSeconds / 3600);
+	const minutes = Math.floor((durationInSeconds % 3600) / 60);
+	const seconds = durationInSeconds % 60;
+
+	if (hours > 3) {
```
Comment (JamieB-gu) [inline dotcom-rendering/src/components/ListenToArticle.importable.tsx]: I think this means anything **4 hours** or more is invalid, rather than 3 hours? In other words, 03:00:00-03:59:59 would still get displayed, but anything longer wouldn't?

If you want anything over 3 hours not to show, as mentioned in the description, I think this would need to be:

```suggestion
	if (hours > 2) {
```

or

```suggestion
	if (hours >= 3) {
```

?

File: dotcom-rendering/src/components/ListenToArticle.importable.tsx [hunk 4]
```diff
@@ -7,26 +7,55 @@ import { ListenToArticleButton } from './ListenToArticleButton';
...
+		hours > 0 ? `${hours.toString().padStart(2, '0')}:` : ''
+	}${minutes > 0 ? minutes.toString().padStart(2, '0') : '00'}:${
+		seconds > 0 ? seconds.toString().padStart(2, '0') : '00'
+	}`;
+
+	return formattedDuration;
+};
 
+export const ListenToArticle = ({ articleId }: Props) => {
+	const [showButton, setShowButton] = useState<boolean>(true);
```
Comment (JamieB-gu) [inline dotcom-rendering/src/components/ListenToArticle.importable.tsx]: This PR describes adding an audio duration, but this default state has also flipped to `true`? I think this means you now initially show the button in all cases, and then hide it if it's not available? Is that intentional?

File: dotcom-rendering/src/components/ListenToArticleButton.tsx [hunk 5]
```diff
@@ -17,16 +17,27 @@ const button = css`
...
 		onClick={onClickHandler}
 		size="small"
 		cssOverrides={button}
 		icon={<SvgMediaControlsPlay />}
 	>
-		Listen to article
+		<span css={durationPadding}>
+			{!!audioDuration && `${audioDuration}`}
+		</span>
+		<span>Listen to article</span>
```
Comment (JamieB-gu) [inline dotcom-rendering/src/components/ListenToArticleButton.tsx]: Do you need this span?

```suggestion
		Listen to article
```

File: dotcom-rendering/src/components/ListenToArticleButton.tsx [hunk 6]
```diff
@@ -17,16 +17,27 @@ const button = css`
...
 	<Button
 		onClick={onClickHandler}
 		size="small"
 		cssOverrides={button}
 		icon={<SvgMediaControlsPlay />}
 	>
-		Listen to article
+		<span css={durationPadding}>
+			{!!audioDuration && `${audioDuration}`}
+		</span>
```
Comment (JamieB-gu) [inline dotcom-rendering/src/components/ListenToArticleButton.tsx]: I think this will add the padding even when `!!audioDuration` resolves to `false`? You may want to make the whole `span` conditional?

```suggestion
		{audioDuration === undefined || audioDuration === '' ? null : (
			<span css={durationPadding}>{audioDuration}</span>
		)}
```

**Review by JamieB-gu — COMMENTED** [review]
Looks good 👍. A few questions/suggestions.

File: dotcom-rendering/src/components/ListenToArticle.test.tsx (hunk 1 above)
Comment (vlbee) [reply dotcom-rendering/src/components/ListenToArticle.test.tsx]: Good shout. "" makes more sense. Just trying to cover any number edge cases.

File: dotcom-rendering/src/components/ListenToArticle.importable.tsx (hunk 2 above)
Comment (vlbee) [reply dotcom-rendering/src/components/ListenToArticle.importable.tsx]: Need to relook at this. I think the final designs I just got don't have the leading zero for single digit values anyways.

File: dotcom-rendering/src/components/ListenToArticle.importable.tsx (hunk 3 above)
Comment (vlbee) [reply dotcom-rendering/src/components/ListenToArticle.importable.tsx]: This was an arbitrary number just to handle an edge case of a very large number. I will check with design. It might be better to cap at 1hr actually.

File: dotcom-rendering/src/components/ListenToArticle.importable.tsx [hunk 7]
```diff
@@ -7,26 +7,51 @@ import { ListenToArticleButton } from './ListenToArticleButton';
...
+export const formatAudioDuration = (durationInSeconds: number): string => {
+	if (durationInSeconds >= 3600 || durationInSeconds <= 0) {
+		return '';
+	}
+	const minutes = Math.floor((durationInSeconds % 3600) / 60);
+	const seconds = durationInSeconds % 60;
+
+	const formattedDuration = `${
+		minutes > 0 ? minutes.toString().padStart(1, '0') : '0'
+	}:${seconds > 0 ? seconds.toString().padStart(2, '0') : '00'}`;
```
Comment (JamieB-gu) [inline dotcom-rendering/src/components/ListenToArticle.importable.tsx]: Do you need the conditionals, or `padStart` for minutes? I think `padStart` will take care of generating `'00'` for you for the seconds, and because you don't want to pad the minutes with zeroes, you don't need the `padStart` method there?

```suggestion
	const formattedDuration = `${
		minutes.toString()
	}:${seconds.toString().padStart(2, '0')}`;
```

Comment (vlbee) [reply dotcom-rendering/src/components/ListenToArticle.importable.tsx]: good shout 👌
